### PR TITLE
testutils/testcluster: explain why we don't remove the leaseholder

### LINKED ISSUE #7872: testutils/testcluster: TestBasicManualReplication slowness
`TestBasicManualReplication` currently tests removal of a non-lease-holder replica. If the test is changed to remove the lease-holder replica, the runtime of the test 8.5x. Need to track down where that slowness is coming from. @bdarnell hypothesizes:

> My guess: at the end of `RemoveReplicas`, you do a consistent read of the range descriptor via the first store. Since that store has a local replica, it tries to read from itself first, and hangs until `SendNextTimeout`. Fixing the races and reducing `SendNextTimeout` would mitigate this problem, but there's probably something better we could do to get the local replica to fail sooner.

### PR DESCRIPTION
**Pull request by petermattis**
Fixes #7872

---

This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/10272)

### CODE REVIEW

**Comment by tamird** [comment]
Hm. Can we shorten the lease duration for this test? Removing the lease holder seems like a worthy case to test.

**Comment by petermattis** [comment]
> Can we shorten the lease duration for this test? Removing the lease holder seems like a worthy case to test.

There is no easy mechanism to shorten the lease duration from `TestCluster`. I did manually whack the values and verified that a shorter lease duration speeds the test up. Removing the lease holder is not a worthy test case as we never do that in real systems (we'd always transfer the lease first).

**Comment by tamird** [comment]
OK. That justification of why we don't test removing the lease holder
should be added to the comment.

LGTM

**Comment by petermattis** [comment]
Comment expanded.

File: pkg/testutils/testcluster/testcluster_test.go [hunk 1]
```diff
@@ -158,8 +158,11 @@ func TestBasicManualReplication(t *testing.T) {
 		t.Fatal(err)
 	}
 
-	// TODO(peter): Removing the range leader (tc.Target(1)) causes the test to
-	// take ~13s vs ~1.5s for removing a non-leader. Track down that slowness.
+	// NB: Removing the leaseholder (tc.Target(1)) causes the test to take ~11s
+	// vs ~1.5s for removing a non-leaseholder. This is due to needing to wait
+	// for the lease to timeout which takes ~9s. Testing leaseholder removal is
+	// not necessary because internal rebalancing avoids every removing the
```
Comment (tamird) [inline pkg/testutils/testcluster/testcluster_test.go]: s/every/ever/

**Comment by tamird** [comment]
LGTM

File: pkg/testutils/testcluster/testcluster_test.go (hunk 1 above)
Comment (petermattis) [inline pkg/testutils/testcluster/testcluster_test.go]: Fixed. Sigh, another round of CI.

**Pull request merged**